ihe-interop/src/main/java/org/motechproject/ihe/interop/service/impl/IHETemplateServiceImpl.java

```diff
@@ -19,14 +21,41 @@
 
     private static final Logger LOGGER = LoggerFactory.getLogger(IHETemplateServiceImpl.class);
 
+    private PostMethod post;
+
+    private HttpClient client;
```

Comment:
Why do we keep those two as class fields, if they are recreated with every method call we make? I don't see a reason they need to live after the method is executed.